### ISSUE TITLE
FEATURE REQUEST: please create a docker image  

### ISSUE THREAD

**Issue by ohld** [postlight/mercury-parser#390]
I think that your tool can be used as a parsing microservice. In that point, I think you need to have a docker image with your tool. In my opinion, that can help to grow your users' community who wanted just to `docker run ... ` and parse.

So you need to provide a `Dockerfile` and upload each stable version into a Dockerhub to allow users to run your service with just one-line `docker pull ...` command.

And sorry to delete all issue template, I didn't find any 'feature request' template.

Thanks. And sorry.

**Comment by NightMachinary** [comment]
It’s just one npm install. npm is itself a one-line installation with most package managers.

**Comment by ohld** [comment]
I understand, but a docker image will allow using your system in a cluster with just `docker- compose` or `kubernetes`. 

Now to run your code the `npm` should be installed on a machine. Also, I don't know if any `npm` version can run your code: I think it is possible that on some `npm` version your code (and just any code) just can't be run. 

On the other hand, if you specify the `docker` image you can select the proper `npm` version to be installed on a container. As a result, your code can be run on any machine who has docker installed - that is the key idea of containerization. 

In other words, if you prepare the docker image, to run the code on any machine only `docker` will be required to be installed. At all scale-ready systems (AWS, Google Cloud etc.) docker is everywhere.

**Comment by NightMachinary** [comment]
Well, why don’t you create a docker image and do a pull request to include
a link to it in the readme? :D
I am not a maintainer here, btw ;)
The code should be useable with the latest version of npm (and node), as it
is under active maintenance. And if it falls out of maintenance, then it’ll
become useless quickly anyhow.

**Comment by fappelman** [comment]
I think a Docker image is more appropriate for the [API version](https://github.com/postlight/mercury-parser-api).
I use this Dockerfile for the AP versionI:

```docker
FROM node:latest

RUN git clone https://github.com/postlight/mercury-parser-api.git
WORKDIR mercury-parser-api
RUN yarn install
EXPOSE 3000
CMD [ "yarn", "serve", "--port", "3000", "--host", "0.0.0.0" ]
```

Maybe you can use this.

**Comment by ohld** [comment]
@fappelman thank you very much! I’ll definitely use your snippet later in my project. I want just a docker container that can parse anything on a web. And this may be a solution. Thanks.

**Comment by ohld** [comment]
@fappelman This API version seems a bit outdated. Is there a way to run a docker with this `mercury-parser` project and call its methods with HTTP requests (like other microservices work)?

**Comment by fappelman** [comment]
The only thing I can advise you is to reinstall the parser after you run "yarn install".
Something like:

```Dockerfile
RUN git clone https://github.com/postlight/mercury-parser-api.git
RUN cd mercury-parser-api ; yarn install
```

This will make sure you use the latest version of the parser. The API is just a front-end.

**Comment by ohld** [comment]
> The API is just a front-end.
Oww, sorry, I just got that this API imports always a fresh copy of `mercury-parser` itself. Thanks!

**Comment by ohld** [comment]
I've created a docker image for it - https://github.com/morejust/mercury-parser-docker
Also, I've uploaded it to dockerhub.

**Comment by fappelman** [comment]
Thanks.
